<launch>

	<!-- Set compression format to png -->
	<param name="/comp/compressed/format" value="png" />
	<param name="/camera/driver/depth_registration" value="true" />	

	  

	<!-- Launch a reconfigure GUI -->
	<node pkg="dynamic_reconfigure" type="reconfigure_gui" name="reconfigure_gui" output="screen" respawn="true"  />
	
	
	<!-- Start the camera-->
	<include file="$(find openni_launch)/launch/openni.launch" />
	<node pkg="rviz" type="rviz" name="rviz_signDetect" output="screen" respawn="true"  />

	<!-- SignDetector -->
	<node pkg="aa_signs" type="signDetection" name="signDetection" output="screen" respawn="true" > 
 		<remap from="/signDetection/depth_registered/image_rect" 	to="/camera/depth_registered/image_rect_raw" />
 		<remap from="/signDetection/rgb/image_rect_color" 			to="/camera/rgb/image_rect_color" />
 		<remap from="/signDetection/rgb/camera_info" 				to="/camera/rgb/camera_info" />
 		<param name="settingsfile" value="(find aa_signs)/settings/signs.xml" />
	</node>

  	<!-- Nodelet for creating a pointcloud out of the detector image (for debuging filters) -->
	<node pkg="nodelet" type="nodelet" name="PointCloudAdvisor" args="manager" output="screen"/>

		<node pkg="nodelet" type="nodelet" name="points_xyzrgb_Advisor" args="load depth_image_proc/point_cloud_xyzrgb PointCloudAdvisor --no-bond">
	    <!-- Explicit topic remappings, shouldn't need all of these -->
	    <remap from="rgb/image_rect_color"        to="/signDetection/out_rgb" />
	    <remap from="rgb/camera_info"             to="/signDetection/camera_info" />
	    <remap from="depth_registered/image_rect" to="/signDetection/out_depth" />
	    <remap from="depth_registered/points"     to="debug_cloud" />
  	</node>
  	
  	
</launch>